spec/javascripts/highlight-active-section-heading-spec.js

```diff
@@ -0,0 +1,92 @@
+/* eslint-env jasmine */
+/* eslint-disable no-multi-str */
+
+describe('A highlight active section heading module', function () {
+  'use strict'
+
+  var module
+
+  beforeEach(function () {
+    module = new GOVUK.Modules.HighlightActiveSectionHeading()
+  })
+
+  afterEach(function () {
+    $(document).off()
+  })
+
+  describe('', function () {
+    var $element = $('<div class="grid-row" data-module="highlight-active-section-heading">\
+      <div class="column-third">\
+        <div class="page-contents js-page-contents js-stick-at-top-when-scrolling">\
+          <h2 class="page-contents__title">Page contents:</h2>\
+          <ul class="page-contents__list">\
+            <li><a href="#section-1">Section 1</a></li>\
+            <li><a href="#section-2">Section 2</a></li>\
+            <li><a href="#section-3">Section 3</a></li>\
+          </ul>\
+        </div>\
+      </div>\
+      <div class="column-two-thirds">\
+        <div class="govspeak-wrapper">\
+          <div class="govuk-govspeak">\
+            <h2 id="section-1">Section 1</h2>\
+            <p>Section 1 text</p>\
+            <h2 id="section-2">Section 2</h2>\
+            <p>Section 2 text</p>\
+            <h2 id="section-3">Section 3</h2>\
+            <p>Section 3 text</p>\
+          </div>\
+        </div>\
+      </div>\
+    </div>')
+
+    describe('On desktop', function () {
+      beforeEach(function () {
+        module.getWindowDimensions = function () {
+          return {
+            height: 768,
+            width: 1024
+          }
+        }
+        module.getHeadingPosition = function (theID) {
+          return {
+            top: 200
+          }
+        }
+        module.getNextHeadingPosition = function (theNextID) {
+          return {
+            top: 400
+          }
+        }
+      })
+
+      it('has no highlighted nav items, when the page loads', function () {
+        module.getWindowPositions = function () {
+          return {
+            scrollTop: 0
+          }
+        }
+        module.start($element)
+        var $anchors = $element.find('.js-page-contents a')
+        expect($anchors.hasClass('active')).toBe(false)
+      })
```

Comment:
Just a style thing for fun.. It's quite popular to break tests into paragraphs - setup, exercise and verification paragraphs. It just means putting a line break in between the paragraphs.

eg.

```
it('has no highlighted nav items, when the page loads', function () {
  setWindowScroll(0)

  module.start($element)

  var $anchors = $element.find('.js-page-contents a')
  expect($anchors.hasClass('active')).toBe(false)
})
```

I also think it'd be cool to hide the complexity of stubbing the window and element position things in a helper function. I had a go in the above example.

None of this essential at all.. Just a suggestion for how I would find it more readable.